WordPress/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift

```diff
@@ -0,0 +1,159 @@
+
+            self.tooltip.alpha = 1
+            tooltipTopConstraint.constant -= Constants.tooltipTopConstraintAnimationOffset
+            self.containerView.layoutIfNeeded()
+        }
+    }
+
+    private func configureDismissal() {
+        tooltip.primaryButtonAction = {
+            UIView.animate(
+                withDuration: Constants.tooltipAnimationDuration,
+                delay: 0,
+                options: .curveEaseOut
+            ) {
+                guard let tooltipTopConstraint = self.tooltipTopConstraint else { return }
+
+                self.tooltip.alpha = 0
+                tooltipTopConstraint.constant += Constants.tooltipTopConstraintAnimationOffset
+                self.containerView.layoutIfNeeded()
+            } completion: { isSuccess in
+                self.primaryTooltipAction?()
+                self.tooltip.removeFromSuperview()
+            }
+        }
+    }
+
+    private func setUpConstraints() {
+        tooltip.translatesAutoresizingMaskIntoConstraints = false
+
+        var tooltipConstraints: [NSLayoutConstraint] = [
+            tooltip.centerXAnchor.constraint(equalTo: containerView.centerXAnchor, constant: extraArrowOffsetX())
+        ]
+
+        switch tooltipOrientation() {
+        case .bottom:
+            tooltipTopConstraint = targetView.topAnchor.constraint(
+                equalTo: tooltip.bottomAnchor,
+                constant: Constants.verticalTooltipDistanceToFocus + Constants.tooltipTopConstraintAnimationOffset
+            )
+        case .top:
+            tooltipTopConstraint = tooltip.topAnchor.constraint(
+                equalTo: targetView.bottomAnchor,
+                constant: Constants.verticalTooltipDistanceToFocus + Constants.tooltipTopConstraintAnimationOffset
+            )
+        }
+
+        tooltipConstraints.append(tooltipTopConstraint!)
+        NSLayoutConstraint.activate(tooltipConstraints)
+    }
+
+    @objc private func recalculatePosition() {
+        tooltip.removeFromSuperview()
+        show()
+    }
+
+    /// Calculates where the arrow needs to place in the borders of the tooltia.
+    /// This depends on the position of the `targetView` relative to `tooltip`.
+    private func arrowOffsetX() -> CGFloat {
+        return targetView.frame.midX - ((containerView.bounds.width - tooltip.size().width) / 2) - extraArrowOffsetX()
+    }
+
+    /// If the tooltip is alwyas vertically centered, tooltip's width may not be big enough to reach to the `targetView`
+    /// If `xxxxxxxx` is the Tooltip and `oo` is the `targetView`:
+    /// |                                               |
+    /// |                xxxxxxxx                 |
+    /// |                                    oo       |
+    /// The tooltip needs an extra X offset to be aligned with target so that tooltip arrow points to the correct position.
+    /// Here the tooltip is pushed to the right so the arrow
```

Comment:
Not sure if I understand this correctly, but perhaps you meant this?
```suggestion
    /// Here the tooltip is pushed to the right so the arrow is aligned to the target.
```